src/confcom/azext_confcom/template_util.py

```diff
@@ -1494,3 +1494,185 @@ def pick(obj, *keys):
+        exec_processes = []
+        old_command_list = old_container.get("execProcesses", [])
+        if old_command_list:
+            exec_processes.append({"command": old_command_list})
+
+        command = old_container.get("command")
+
+        # Liveness probe => exec process
+        liveness_probe = old_container.get("livenessProbe", {})
+        liveness_exec = liveness_probe.get("exec", {})
+        liveness_command = liveness_exec.get("command", [])
+        if liveness_command:
+            exec_processes.append({
+                "command": liveness_command
+            })
+
+        # Readiness probe => exec process
+        readiness_probe = old_container.get("readinessProbe", {})
+        readiness_exec = readiness_probe.get("exec", {})
+        readiness_command = readiness_exec.get("command", [])
+        if readiness_command:
+            exec_processes.append({
+                "command": readiness_command
+            })
+
+
+        # Build the 'volumeMounts' section
+        volume_mounts = []
+        for mount in old_container.get("mounts") or []:
+            # For 'name', we can take the mountType or generate something else:
+            # e.g. if mountType is "azureFile", name "azurefile"
+            mount_name = mount.get("mountType", "defaultName").lower()
+            volume_mount = {
+                "name": mount_name,
+                "mountPath": mount.get("mountPath"),
+                "mountType": mount.get("mountType"),
+                "readOnly": mount.get("readonly", True),
+            }
+            volume_mounts.append(volume_mount)
+
+        # Create the container's "properties" object
+        container_properties = {
+            "image": old_container.get("containerImage"),
+            "execProcesses": exec_processes,
+            "volumeMounts": volume_mounts,
+            "environmentVariables": new_envs,
+            "command": command,
+        }
+
+        if old_container.get("securityContext") is not None:
+            container_properties["securityContext"] = old_container["securityContext"]
+
+        if old_container.get("allow_elevated") is not None:
+            if "securityContext" not in container_properties:
+                container_properties["securityContext"] = {}
+            container_properties["securityContext"]["privileged"] = old_container.get("allow_elevated")
+
+        if old_container.get("workingDir") is not None:
+            container_properties["workingDir"] = old_container.get("workingDir")
+
+        # Finally, assemble the new container dict
+        new_container = {
+            "name": old_container.get("name"),
+            "properties": container_properties
+        }
+
+        new_data["containers"].append(new_container)
+
+    return new_data
+
+
+def detect_old_format(old_data):
+    old_containers = old_data.get("containers", [])
+    if len(old_containers) > 0 and old_containers[0].get(config.ACI_FIELD_CONTAINERS_CONTAINERIMAGE) is not None:
```

Comment:
Based on your commented example, this could be done by checking if fragments exist... Since I'm assuming not, you should comment that in the example

Reply:
Not every config file in the new format is going to import fragments. I don't think that would be a valid check

Reply:
Yeah I figured that was the case, so make a comment about it so that anyone who may look later understands why you're doing this check